src/receivers/custom-routes.ts

```diff
@@ -0,0 +1,48 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import { CustomRouteInitializationError } from '../errors';
+
+export interface CustomRoute {
+  path: string;
+  method: string | string[];
+  handler: (req: IncomingMessage, res: ServerResponse) => void;
+}
+
+export interface ReceiverRoutes {
+  [key: string]: {
+    [key: string]: (req: IncomingMessage, res: ServerResponse) => void;
+  };
+}
```

Comment:
nit: how about having a bit clearer key names for slightly better readability?

```suggestion
export interface ReceiverRoutes {
  [url: string]: {
    [method: string]: (req: IncomingMessage, res: ServerResponse) => void;
  };
}
```